### ISSUE TITLE
How can one use newly added GPTJ6B model for finetuning?

### ISSUE THREAD

**Issue by working12** [huggingface/transformers#13841]
I want to finetune GPTJ6B for summarization and question answering. Are there any guidelines on how to do that?

**Comment by NielsRogge** [comment]
From the [docs](https://huggingface.co/transformers/master/model_doc/gptj.html):

> The model should fit on 16GB GPU for inference. For training/fine-tuning it would take much more GPU RAM. Adam optimizer for example makes four copies of the model: model, gradients, average and squared average of the gradients. So it would need at least 4x model size GPU memory, even with mixed precision as gradient updates are in fp32. This is not including the activations and data batches, which would again require some more GPU RAM. So one should explore solutions such as DeepSpeed, to train/fine-tune the model. Another option is to use the original codebase to train/fine-tune the model on TPU and then convert the model to Transformers format for inference. Instructions for that could be found [here](https://github.com/kingoflolz/mesh-transformer-jax/blob/master/howto_finetune.md).

**Comment by working12** [comment]
@NielsRogge The problem is original repo uses TPU so one needs Cloud storage. (Not feasible for many people). That's why I was trying to do it with huggingface repo. The section you pointed out, the requirement about RAM isn't it possible to run in Colab Pro? If yes I just want to fine tune with QA and Summarization task separately. I am looking for some concrete code example on how to get started with that.

**Comment by NielsRogge** [comment]
Unfortunately, Colab Pro is not enough, as you need at least 108 GB of GPU RAM (this is when using DeepSpeed). You can estimate the memory required using the [following utility](https://deepspeed.readthedocs.io/en/stable/memory.html):

```
$ python -c 'from transformers import AutoModel; \
from deepspeed.runtime.zero.stage3 import estimate_zero3_model_states_mem_needs_all_live; \
model = AutoModel.from_pretrained("EleutherAI/gpt-j-6B"); \
estimate_zero3_model_states_mem_needs_all_live(model, num_gpus_per_node=1, num_nodes=1)'
Estimated memory needed for params, optim states and gradients for a:
HW: Setup with 1 node, 1 GPU per node.
SW: Model with 5844M total params, 206M largest layer params.
  per CPU  |  per GPU |   Options
  146.96GB |   0.77GB | cpu_offload=1, cpu_offload_params=1, zero_init=1
  146.96GB |   0.77GB | cpu_offload=1, cpu_offload_params=1, zero_init=0
  130.63GB |  11.66GB | cpu_offload=1, cpu_offload_params=0, zero_init=1
  130.63GB |  11.66GB | cpu_offload=1, cpu_offload_params=0, zero_init=0
    1.15GB |  98.74GB | cpu_offload=0, cpu_offload_params=0, zero_init=1
   32.66GB |  98.74GB | cpu_offload=0, cpu_offload_params=0, zero_init=0
```
